pkg/apis/monitoring/v1/servicemonitor_types.go

```diff
@@ -178,6 +183,40 @@ type ServiceMonitorSpec struct {
 	BodySizeLimit *ByteSize `json:"bodySizeLimit,omitempty"`
 }
 
+// ServiceMonitorStatus is the most recent observed status of the ServiceMonitor. Read-only.
+// More info:
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +k8s:openapi-gen=true
+type ServiceMonitorStatus struct {
+	// The list of resources that the service monitor is bound to.
+	// +patchMergeKey=resource
+	// +patchMergeKey=namespace
+	// +patchMergeKey=name
+	// +patchMergeStrategy=merge
+	// +optional
+	Bindings []ServiceMonitorBinding `json:"bindings,omitempty"`
+}
+
+// ServiceMonitorBinding is a binding of a ServiceMonitor to a resource.
+// +k8s:openapi-gen=true
+type ServiceMonitorBinding struct {
+	// The type of resource being referenced (e.g. Prometheus or PrometheusAgent).
+	// +kubebuilder:validation:Enum=prometheuses;prometheusagents
+	// +required
+	Resource string `json:"resource"`
+	// The name of the referenced object.
+	// +required
```

Comment:
kubebuilder:validation:MinLength=1 
```suggestion
	// +required
	// +kubebuilder:validation:MinLength=1
```


I'm torn since this will not be manually created, but would be useful.

Reply:
done